platform/mellanox/mlnx-platform-api/sonic_platform/module.py

```diff
@@ -247,3 +249,182 @@ def get_sfp(self, index):
+        """
+        return ModuleBase.MODULE_TYPE_DPU
+
+    def get_name(self):
+        """
+        Retrieves the name of the module prefixed by SUPERVISOR, LINE-CARD,
+        FABRIC-CARD, SWITCH, DPU0, DPUX
+
+        Returns:
+            A string, the module name prefixed by one of MODULE_TYPE_SUPERVISOR,
+            MODULE_TYPE_LINE or MODULE_TYPE_FABRIC or MODULE_TYPE_DPU or
+            MODULE_TYPE_SWITCH and followed by a 0-based index.
+
+            Ex. A Chassis having 1 supervisor, 4 line-cards and 6 fabric-cards
+            can provide names SUPERVISOR0, LINE-CARD0 to LINE-CARD3,
+            FABRIC-CARD0 to FABRIC-CARD5.
+            A SmartSwitch having 4 DPUs and 1 Switch can provide names DPU0 to
+            DPU3 and SWITCH
+        """
+        return self._name
+
+    def get_description(self):
+        """
+        Retrieves the platform vendor's product description of the module
+
+        Returns:
+            A string, providing the vendor's product description of the module.
+        """
+        return "NVIDIA BlueField-3 DPU"
+
+    def get_oper_status(self):
+        if self.fault_state:
+            return ModuleBase.MODULE_STATUS_FAULT
+        if utils.read_int_from_file(f'/run/hw-management/events/dpu{self.dpu_id}_ready') == 1:
+            return ModuleBase.MODULE_STATUS_ONLINE
+        elif utils.read_int_from_file(f'/run/hw-management/events/dpu{self.dpu_id}_shtdn_ready') == 1:
+            return ModuleBase.MODULE_STATUS_OFFLINE
+        return ModuleBase.MODULE_STATUS_FAULT # If both ready and shtdn_ready are not set, return fault state
+
+    ##############################################
+    # SmartSwitch methods
+    ##############################################
+
+    def get_dpu_id(self):
+        """
+        Retrieves the DPU ID. Returns None for non-smartswitch chassis.
+        Returns:
+            An integer, indicating the DPU ID. DPU0 returns 1, DPUX returns X+1
+            Returns '0' on switch module
+        """
+        return self.dpu_id
+
+    def get_reboot_cause(self):
+        """
+        Retrieves the cause of the previous reboot of the DPU module
+        Returns:
+            A tuple (string, string) where the first element is a string
+            containing the cause of the previous reboot. This string must
+            be one of the predefined strings in this class. If the first
+            string is "REBOOT_CAUSE_HARDWARE_OTHER", the second string can be
+            used to pass a description of the reboot cause.
+            Some more causes are appended to the existing list to handle other
+            modules such as DPUs.
+            Ex: REBOOT_CAUSE_POWER_LOSS, REBOOT_CAUSE_HOST_RESET_DPU,
+            REBOOT_CAUSE_HOST_POWERCYCLED_DPU, REBOOT_CAUSE_SW_THERMAL,
+            REBOOT_CAUSE_DPU_SELF_REBOOT
+        """
+        if utils.read_int_from_file(f'/run/hw-management/system/dpu{self._name.lower()}/system/reset_from_main_board') == 1:
```

Comment:
instead of having a long if-else statement can we create a map from the file to the reboot cause? E.g.:
```
reboot_cause_map = {
    f'/run/hw-management/system/dpu{self._name.lower()}/system/reset_from_main_board': ModuleBase.REBOOT_CAUSE_HOST_RESET_DPU,
    f'/run/hw-management/system/dpu{self._name.lower()}/system/reset_dpu_thermal': ModuleBase.REBOOT_CAUSE_SW_THERMAL
}
for f, rc in reboot_cause_map.items():
     if utils.read_int_from_file(f) == 1:
            return rc, ''
return ModuleBase.REBOOT_CAUSE_NON_HARDWARE, ''
```

Reply:
Done